syntax = "proto3";

message IndexSpec {
    string name = 1;
    repeated PathSpec paths = 2 [json_name = "fs_paths"];
    repeated RepoSpec repositories = 3 [json_name = "repositories"];
}

message Metadata {
    string url_pattern = 1     [json_name = "url_pattern"];
    string remote = 2          [json_name = "remote"];
    string github = 3          [json_name = "github"];
    repeated string labels = 4 [json_name = "labels"];
}

message CloneOptions {
    int32 depth = 1            [json_name = "depth"];
    string username = 2        [json_name = "username"];
    string password_env = 3    [json_name = "password_env"];
}

message PathSpec {
    string path = 1             [json_name = "path"];
    string name = 2             [json_name = "name"];
    string ordered_contents = 3 [json_name = "ordered_contents"];
    Metadata metadata = 4       [json_name = "metadata"];
    bool ignore_symlinks = 5    [json_name = "ignore_symlinks"];
}

message RepoSpec {
    string path = 1                [json_name = "path"];
    string name = 2                [json_name = "name"];
    repeated string revisions = 3  [json_name = "revisions"];
    Metadata metadata = 4          [json_name = "metadata"];
    bool walk_submodules = 5       [json_name = "walk_submodules"];
    CloneOptions clone_options = 6 [json_name = "clone_options"];
}
